### ISSUE TITLE
prepareGhPages deletes files in the root directory instead of those in the target directory

### ISSUE THREAD

**Issue by siordache** [ajoberstar/gradle-git#196]
A preliminary action of the prepareGhPages task consists in deleting all files in the target directory (that is, the directory configured using `into`).
However, this doesn't work as expected: the current implementation creates a list containing the names of the files found in the target directory and then removes the corresponding files found in the root of the gh-pages branch of the repositiory.
Moreover, files in subdirectories of the target directory are ignored.

I will send a PR that fixes this issue.

**Comment by siordache** [comment]
Resolved by #193.

**Comment by siordache** [comment]
I'm reopening this issue, because it has been only partially solved.

The problem is that in the [code below](https://github.com/ajoberstar/gradle-git/blob/7fdd4abdbf404aee696bc1079f6d82ad40c09bef/src/main/groovy/org/ajoberstar/gradle/git/ghpages/GithubPagesPlugin.groovy#L100-L104) `filesList` contains only file names instead of relative paths.
```
def targetDir = new File(extension.workingDir, extension.pages.relativeDestinationDir)
def filesList = targetDir.list { dir, name -> !name.equals('.git') }
if (filesList && extension.deleteExistingFiles) {
    repo.remove(patterns: filesList)
}
```
To fix this, each file name in `filesList` should be prefixed with `extension.pages.relativeDestinationDir`.
I will send a PR.
